Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -936,6 +942,73 @@ public final class Chat {
         try await channelUpdater.stopWatching(cid: cid)
     }
     
+    /// Subscribes to web-socket events of a single type in this channel.
+    ///
+    /// - Note: The handler is always called on the main thread.
+    ///
+    /// - Parameters:
+    ///   - event: The event type to subscribe to (e.g. ``MessageNewEvent``).
+    ///   - handler: The handler closure which is called when the event happens.
+    ///
+    /// - Returns: A cancellable instance, which you use when you end the subscription. Deallocation of the result will tear down the subscription stream.
+    public func subscribe<T>(to event: T.Type, handler: @escaping (T) -> Void) -> AnyCancellable where T: Event {
+        eventNotificationCenter.subscribe(
+            to: event,
+            filter: { [cid] in
+                EventNotificationCenter.channelFilter(cid: cid, event: $0)
+            },
+            handler: handler
+        )
+    }
+    
+    /// Subscribes to web-socket events of a single type in this channel.
+    ///
+    /// - Parameter event: The event type to subscribe to (e.g. ``MessageNewEvent``).
+    ///
+    /// - Returns: Web-socket events of this channel, as an asynchronous stream of events.
+    public func subscribe<T>(to event: T.Type) -> AsyncStream<T> where T: Event {
```

Comment:
Topic for discussion, but I would actually prefer using only Combine cancellables - the room for error is too big with async streams. Curious to hear the others opinion.

Reply:
Yeah, probably makes sense to get rid of the stream variant (at least for now). One can always create streams later with the handler (if someone really wants to use streams).
@nuno-vieira please also share your opinion.

Reply:
Let me remove it for now and we can consider adding it back if we decide to do so.